pkg/controllermanager/apis/config/v1alpha1/defaults.go

```diff
@@ -18,331 +18,335 @@ import (
 	}
-}
-
-// SetDefaults_SeedBackupBucketsCheckControllerConfiguration sets defaults for the given SeedBackupBucketsCheckControllerConfiguration.
-func SetDefaults_SeedBackupBucketsCheckControllerConfiguration(obj *SeedBackupBucketsCheckControllerConfiguration) {
-	if obj.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.ConcurrentSyncs = &v
+	if obj.ControllerRegistration == nil {
+		obj.ControllerRegistration = &ControllerRegistrationControllerConfiguration{}
 	}
-	if obj.SyncPeriod == nil {
-		obj.SyncPeriod = &metav1.Duration{Duration: 30 * time.Second}
+	if obj.ExposureClass == nil {
+		obj.ExposureClass = &ExposureClassControllerConfiguration{}
 	}
-}
-
-// SetDefaults_CertificateSigningRequestControllerConfiguration sets defaults for the given CertificateSigningRequestControllerConfiguration.
-func SetDefaults_CertificateSigningRequestControllerConfiguration(obj *CertificateSigningRequestControllerConfiguration) {
-	if obj.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.ConcurrentSyncs = &v
+	if obj.Project == nil {
+		obj.Project = &ProjectControllerConfiguration{}
 	}
-}
-
-// SetDefaults_SeedControllerConfiguration sets defaults for the given SeedControllerConfiguration.
-func SetDefaults_SeedControllerConfiguration(obj *SeedControllerConfiguration) {
-	if obj.SyncPeriod == nil {
-		obj.SyncPeriod = &metav1.Duration{Duration: 10 * time.Second}
+	if obj.Quota == nil {
+		obj.Quota = &QuotaControllerConfiguration{}
 	}
-
-	if obj.ConcurrentSyncs == nil {
-		v := DefaultControllerConcurrentSyncs
-		obj.ConcurrentSyncs = &v
+	if obj.SecretBinding == nil {
+		obj.SecretBinding = &SecretBindingControllerConfiguration{}
 	}
-
-	if obj.MonitorPeriod == nil {
-		v := metav1.Duration{Duration: 40 * time.Second}
-		obj.MonitorPeriod = &v
+	if obj.Seed == nil {
+		obj.Seed = &SeedControllerConfiguration{}
+	}
+	if obj.SeedExtensionsCheck == nil {
+		obj.SeedExtensionsCheck = &SeedExtensionsCheckControllerConfiguration{}
+	}
+	if obj.SeedBackupBucketsCheck == nil {
+		obj.SeedBackupBucketsCheck = &SeedBackupBucketsCheckControllerConfiguration{}
+	}
+	if obj.ShootQuota == nil {
+		obj.ShootQuota = &ShootQuotaControllerConfiguration{}
+	}
+	if obj.ShootReference == nil {
+		obj.ShootReference = &ShootReferenceControllerConfiguration{}
+	}
+	if obj.ShootRetry == nil {
+		obj.ShootRetry = &ShootRetryControllerConfiguration{}
+	}
+	if obj.ShootConditions == nil {
+		obj.ShootConditions = &ShootConditionsControllerConfiguration{}
+	}
+	if obj.ShootStatusLabel == nil {
+		obj.ShootStatusLabel = &ShootStatusLabelControllerConfiguration{}
 	}
 
-	if obj.ShootMonitorPeriod == nil {
-		v := metav1.Duration{Duration: 5 * obj.MonitorPeriod.Duration}
-		obj.ShootMonitorPeriod = &v
+	if obj.ManagedSeedSet == nil {
+		obj.ManagedSeedSet = &ManagedSeedSetControllerConfiguration{
+			SyncPeriod: metav1.Duration{
+				Duration: 30 * time.Minute,
+			},
```

Comment:
Move this inside `SetDefaults_ManagedSeedSetControllerConfiguration`

Reply:
The problem is that the sync period is being defaulted to 30 minutes only if ManagedSeedSet has not been set. If we move it to `SetDefaults_ManagedSeedSetControllerConfiguration` then it will be defaulted to 30 minutes in cases of ManagedSeedSet being set to non-nil too